app/src/main/java/com/hedvig/app/feature/insurance/ui/tab/InsuranceFragment.kt

```diff
@@ -75,7 +75,7 @@ class InsuranceFragment : Fragment(R.layout.fragment_insurance) {
             ?.also { insuranceViewModel.crossSellActionOpened() }
 
           errorMessage?.let {
-            adapter?.submitList(listOf(InsuranceModel.Header, InsuranceModel.Error))
+            adapter?.submitList(listOf(InsuranceModel.Header, InsuranceModel.Error(it)))
           }
 
           items?.let {
```

Comment:
In this entire section we make it so that if somehow we get both errorMessage as non-null and items as non-null (not sure if possible) we would override the other thing by submitting a new list again. Could we maybe make this a when statement or something like that so that only one of all of them is run? With order of importance like first check if the errors are not null, show that and so on. Or something like that.
Or is there some other reason we're not doing that here that I may be missing?